kairon/shared/data/processor.py

```diff
@@ -1132,6 +1160,117 @@ def __prepare_training_story_events(self, events, timestamp):
             elif event.type == SlotSet.type_name:
                 yield SlotSet(key=event.name, value=event.value, timestamp=timestamp)
 
+    def __extract_multiflow_story_events_steps(self, step_data, bot, user):
+        step_event = {}
+        component_id = None
+        try:
+            if step_data:
+                if step_data.get('type') == StoryStepType.intent.value:
+                    intent = Intents.objects(bot=bot).get(name=step_data.get('name'))
+                    component_id = intent.id.__str__()
+                    step_event['type'] = step_data.get('type')
+                    step_event['name'] = step_data.get('name')
+                    step_event['component_id'] = component_id
+                elif step_data.get('type') == StoryStepType.slot.value:
+                    slot = Slots.objects(name=step_data.get('name'), bot=bot).get()
+                    component_id = slot.id.__str__()
+                    step_event['type'] = step_data.get('type')
+                    step_event['name'] = step_data.get('name')
+                    step_event['value'] = step_data.get('value')
+                    step_event['component_id'] = component_id
+                else:
+                    if step_data.get('name').startswith('utter_'):
+                        utterance = Utterances.objects(bot=bot).get(name=step_data.get('name'))
+                        component_id = utterance.id.__str__()
+                        step_event['type'] = step_data.get('type')
+                        step_event['name'] = step_data.get('name')
+                        step_event['component_id'] = component_id
+                    elif step_data.get('type') in StoryStepType.__members__.values():
```

Comment:
this should be taken from particular action type instead.
for egL if step_type is HTTP_ACTION, the retrieve http action.